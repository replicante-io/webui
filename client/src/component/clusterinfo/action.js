'use strict';
//@flow


/** Name actions for type checking. */
//export const CLUSTERS_LIST = 'CLUSTERS_LIST';


/** Type enum of all possible fetch actions. */
//export type ListClustersAction = {
//  +type: typeof CLUSTERS_LIST,
//  +clusters: Array<ClusterMeta>,
//};

export type ClusterInfoAction =
  {type: 'FLOW_CATCH_ALL'};
