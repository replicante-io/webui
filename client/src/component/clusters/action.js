'use strict';
//@flow

import type { ClusterMeta } from '../dashboard/action';


/** Name actions for type checking. */
export const CLUSTERS_LIST = 'CLUSTERS_LIST';
export const CLUSTERS_SEARCH = 'CLUSTERS_SEARCH';


/** Type enum of all possible fetch actions. */
export type ListClustersAction = {
  +type: typeof CLUSTERS_LIST,
  +clusters: Array<ClusterMeta>,
};

export type SearchClustersAction = {
  +type: typeof CLUSTERS_SEARCH,
  +search: string,
};

export type ClustersAction =
  ListClustersAction |
  SearchClustersAction |
  {type: 'FLOW_CATCH_ALL'};


export function searchClusters(search: string): SearchClustersAction {
  return {
    type: CLUSTERS_SEARCH,
    search: search,
  };
}
