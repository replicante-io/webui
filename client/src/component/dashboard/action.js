'use strict';
//@flow


/** Name actions for type checking. */
export const DASHBOARD_CLUSTERS_SET = 'DASHBOARD_CLUSTERS_SET';


/** Types used by actions. */
export type ClusterMeta = {
  agents_down: number,
  cluster_display_name: string,
  cluster_id: string,
  kinds: Array<string>,
  nodes: number,
  nodes_down: number,
  shards_count: number,
  shards_primaries: number,
};


/** Type enum of all possible fetch actions. */
export type StoreClustersAction = {
  +type: typeof DASHBOARD_CLUSTERS_SET,
  +clusters: Array<ClusterMeta>,
};

export type DashboardAction = 
  StoreClustersAction |
  {type: 'FLOW_CATCH_ALL'};


export function storeClusters(clusters: Array<ClusterMeta>): StoreClustersAction {
  return {
    type: DASHBOARD_CLUSTERS_SET,
    clusters: clusters,
  }
}
