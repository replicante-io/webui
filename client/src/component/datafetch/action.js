'use strict';
//@flow


/** Name actions for type checking. */
export const FETCH_COMPLETE = 'FETCH_COMPLETE';
export const FETCH_DATA = 'FETCH_DATA';


/** Type enum of all possible fetch actions. */
export type FetchComplete = {
  +type: typeof FETCH_COMPLETE,
  +id: String,
}

export type FetchData = {
  +type: typeof FETCH_DATA,
  +id: String,
}

export type FetchAction = 
  FetchComplete |
  FetchData |
  {type: 'FLOW_CATCH_ALL'};
