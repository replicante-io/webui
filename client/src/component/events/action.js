'use strict';
//@flow


export type Event = {
  event: string,
  payload: any,
  timestamp: string,
};


/** Name actions for type checking. */
export const EVENTS_FETCH = 'EVENTS_FETCH';
export const EVENTS_STORE = 'EVENTS_STORE';


/** Type enum of all possible fetch actions. */
export type EventsFetchAction = {
  +type: typeof EVENTS_FETCH,
};

export type EventsStoreAction = {
  +type: typeof EVENTS_STORE,
  +events: Array<Event>,
};

export type EventsAction =
  EventsFetchAction |
  EventsStoreAction |
  {type: 'FLOW_CATCH_ALL'};


export function fetchEvents(): EventsFetchAction {
  return {
    type: EVENTS_FETCH
  };
}


export function storeEvents(events: Array<Event>): EventsStoreAction {
  return {
    type: EVENTS_STORE,
    events: events,
  };
}
