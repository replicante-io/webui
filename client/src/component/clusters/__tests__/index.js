'use strict';

import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { CLUSTERS_SEARCH } from '../action';
import { InnerClusters } from '../index';
import Clusters from '../index';


const mockStore = configureStore([]);
const CLUSTERS = [{
  kinds: ['MongoDB'],
  name: 'test1',
  nodes: 1
}, {
  kinds: ['Kafka'],
  name: 'test2',
  nodes: 4
}];


describe('Clusters', () => {
  describe('index', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusters: {
          clusters: [],
          search: '',
        }
      });
    });

    test('renders no clusters', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <Clusters />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders with clusters', () => {
      let store = mockStore({
        clusters: {
          clusters: CLUSTERS,
          search: '',
        }
      });
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <Clusters />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('search clusters', () => {
      jest.useFakeTimers();
      let component = ReactTestUtils.renderIntoDocument(
        <InnerClusters clusters={[]} dispatch={store.dispatch} search={''} />
      );
      let node = component.input;
      node.value = 'test';
      ReactTestUtils.Simulate.change(node);
      jest.runOnlyPendingTimers();

      const actions = store.getActions();
      expect(actions).toEqual([{
        type: CLUSTERS_SEARCH,
        search: '',
      }, {
        type: CLUSTERS_SEARCH,
        search: 'test',
      }]);
    });

    test('search clusters only once', () => {
      jest.useFakeTimers();
      let component = ReactTestUtils.renderIntoDocument(
        <InnerClusters clusters={[]} dispatch={store.dispatch} search={''} />
      );
      let node = component.input;
      node.value = 'te';
      ReactTestUtils.Simulate.change(node);
      node.value = 'test';
      ReactTestUtils.Simulate.change(node);
      jest.runOnlyPendingTimers();

      const actions = store.getActions();
      expect(actions).toEqual([{
        type: CLUSTERS_SEARCH,
        search: '',
      }, {
        type: CLUSTERS_SEARCH,
        search: 'test',
      }]);
    });

  });
});
