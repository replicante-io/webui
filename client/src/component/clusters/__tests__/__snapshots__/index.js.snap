// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clusters index renders no clusters 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Clusters
        <input
          className="cluster-search"
          defaultValue=""
          onChange={[Function]}
          placeholder="Search ..."
          type="text"
        />
      </div>
      <div
        className="card-body text-center"
      >
        <p
          className="card-text large-icon"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </p>
        <p
          className="card-text"
        >
          No clusters were found.
          <br />
          Try searching something else.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Clusters index renders with clusters 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Clusters
        <input
          className="cluster-search"
          defaultValue=""
          onChange={[Function]}
          placeholder="Search ..."
          type="text"
        />
      </div>
      <div
        className="card-body text-center"
      >
        <div
          className="row cluster-item"
        >
          <div
            className="col"
          >
            <p>
              <a
                href="/clusters/test1"
                onClick={[Function]}
              >
                test1
              </a>
            </p>
            <small>
              Name
            </small>
          </div>
          <div
            className="col"
          >
            <p>
              1
            </p>
            <small>
              Nodes
            </small>
          </div>
          <div
            className="col"
          >
            <p>
              <span
                className="db-icon mongodb"
                title="MongoDB"
              />
            </p>
            <small>
              MongoDB
            </small>
          </div>
        </div>
        <div
          className="row cluster-item"
        >
          <div
            className="col"
          >
            <p>
              <a
                href="/clusters/test2"
                onClick={[Function]}
              >
                test2
              </a>
            </p>
            <small>
              Name
            </small>
          </div>
          <div
            className="col"
          >
            <p>
              4
            </p>
            <small>
              Nodes
            </small>
          </div>
          <div
            className="col"
          >
            <p>
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-database fa-w-14 "
                data-icon="database"
                data-prefix="fas"
                role="img"
                style={Object {}}
                title="Kafka"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 73.143v45.714C448 159.143 347.667 192 224 192S0 159.143 0 118.857V73.143C0 32.857 100.333 0 224 0s224 32.857 224 73.143zM448 176v102.857C448 319.143 347.667 352 224 352S0 319.143 0 278.857V176c48.125 33.143 136.208 48.572 224 48.572S399.874 209.143 448 176zm0 160v102.857C448 479.143 347.667 512 224 512S0 479.143 0 438.857V336c48.125 33.143 136.208 48.572 224 48.572S399.874 369.143 448 336z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </p>
            <small>
              Kafka
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
